### ISSUE TITLE
Simple build of freshly cloned repository modifies repository files!

### ISSUE THREAD

**Issue by Fish-Git** [SDL-Hercules-390/hyperion#403]
After performing a simple build of a freshly cloned repository, git now shows some repository files as having been modified:

```
neon@vm-kdeneon-520:~/hercules/hercules-0$ git ls-files -m
libltdl/config.h
libltdl/libtool
neon@vm-kdeneon-520:~/hercules/hercules-0$ 
```

Why is this?

Bill? &nbsp; (@wrljet) &nbsp; What's going on, buddy?

If building Hercules now always rebuilds libtool each time (which I'm presuming is why config.h is listed too in addition to libtool), then it seems to me that _neither_  `libltdl/libtool` nor `libltdl/config.h` should be part of the repository.

I think we need to remove these two files from the repository (the below is just a guess; I don't do git from the command line; I use TortoiseGit on Windows instead):

```
git rm --cached libltdl/config.h
git rm --cached libltdl/libtool
git commit -m "remove libltdl/libtool and libltdl/config.h from repository"
git push HEAD master
```

Yes? HELP!

**Comment by giskard45** [comment]
Does not happen on on apple Big Sur, for both architectures, doing an out of source build 

config.h and libtool are generated in both the main build directory and in the libltdl subdirectory

but I agree that config.h and libtool should not be distrinuted there to start with 

enrico

**Comment by wrljet** [comment]
Yes, they got checked in by mistake, and I am going to remove those two files.

They are created by `configure`.

And yes, out-of-source building is preferable.  When I update the build instructions, I'll add a note about that.

Bill

**Comment by giskard45** [comment]
just to be sureI just ran an IN SOURCE build and ...
libltdl/config.h was NOT  changed and the content was the same as the one from an out of source build
libltdl/libtool WAS changed because I run the tests on Big Sur  11.4 on the arm processor, Bill created the whole thing on on Big Sur  11.3 with an intel processor

sadly Fish fell in his own trap of not telling enough :-)
it would have been nice to know if He was running an IN vs. an OUT of source build 

e

**Comment by wrljet** [comment]
> 
> If building Hercules now always rebuilds libtool each time (which I'm presuming is why config.h is listed too in addition to libtool), then it seems to me that _neither_ `libltdl/libtool` nor `libltdl/config.h` should be part of the repository.
> 

Done

**Comment by Fish-Git** [comment]
> Sadly, Fish fell in his own trap of not telling enough. &nbsp; `:-)`
> It would have been nice to know if he was running an IN vs. an OUT of source build.

Well....

IMO the commonly accepted "normal" (typical) build procedure is in-source: after cloning you point to the just cloned directory (where your source code is for what you're wanting to build) and do your "build" (./configure and make) from there.

FWIW it's been done that same way on Windows too since forever. This is controlled via your project's "build/output" directory setting, which defaults to being a subdirectory within the source tree. But doing an out-of-source build is as easy as pointing that variable somewhere else instead. It's just that nobody ever does it, since it makes more sense to keep the resulting output binaries with the source code that was used to create them.

Besides, removing them from the source tree (thereby returning your source tree to its original pristine state) is as simple as simply moving that "output" directory to somewhere else (or deleting it altogether). So all things considered it's really no big deal anyway. &nbsp; _&lt;shrug&gt;_

Bottom line: out of source builds are not "normal" in the world I come from, in-source builds are, which is the reason why I didn't bother mentioning it in my report.

**Comment by Fish-Git** [comment]
Closing.
